### PR TITLE
Horizon updates b1

### PR DESCRIPTION
**Pull request by Garyfallidis**
Testing mesa and xvfb issues

### CODE REVIEW

File: dipy/viz/app.py [hunk 1]
```diff
@@ -465,188 +451,264 @@ def win_callback(obj, event):
...
-                                        VisibilityOn()
-                                    c.VisibilityOff()
-                                    self.cea[c]['expanded'] = 1
+                if key == 'r' or key == 'R':
+                    reset()
+
+        options = [r'un\hide centroids', 'invert selection',
+                   r'un\select all', 'expand clusters',
+                   'collapse clusters', 'save streamlines',
+                   'recluster']
```
Comment (skoudoro) [inline dipy/viz/app.py]: 'recluster' option is not visible on the panel. Is it voluntary?

File: dipy/viz/app.py [hunk 2]
```diff
@@ -465,188 +451,264 @@ def win_callback(obj, event):
...
-                            if not self.cea[c]['expanded']:
-                                len_ = self.cea[c]['length']
-                                sz_ = self.cea[c]['size']
-                                if (len_ >= self.length_min and
-                                        sz_ >= self.size_min):
-                                    self.cea[c]['cluster_actor']. \
-                                        VisibilityOn()
-                                    c.VisibilityOff()
-                                    self.cea[c]['expanded'] = 1
+                if key == 'r' or key == 'R':
```
Comment (skoudoro) [inline dipy/viz/app.py]: `reset()` unhide hidden clusters?  it will be good to collapse only the visible cluster

**Comment by skoudoro** [comment]
`dipy_horizon --cluster --cluster_thr 20 --random_colors C:/Users/skoudoro/.dipy/bundle_atlas_hcp842/Atlas_80_Bundles/whole_brain/whole_brain_MNI.trk` give me only 1 colormap as you see below. I think the `random_colors` as a issue or it is normal ?

image

**Comment by skoudoro** [comment]
This PR looks good to me.

If there is no other comment until wednesday I will go ahead and merge it.

**Comment by arokem** [comment]
@skoudoro : I see that there are still a couple of comments you made that are not addressed.

**Comment by skoudoro** [comment]
Indeed, but I believe now that they deserve a new, smallest and focus PR.
@Garyfallidis will have to create a bunch of small PR. Maybe I should create an issue for it

**Comment by skoudoro** [comment]
thank you @Garyfallidis! merging and waiting for new PR

**Pull request merged**